react/components/Promotion/Conditions/Objects/SkusSelectObject/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from 'react'
+import { useIntl } from 'react-intl'
+import { useQuery } from 'react-apollo'
+import uniqBy from 'ramda/src/uniqBy'
+import { Tooltip } from 'vtex.styleguide'
+import { components, MultiValueProps, OptionProps } from 'react-select'
+
+import withSkus, {
+  Sku,
+  WithSkusProps,
+} from '../../../../../connectors/withSkus'
+import SkuBulkInsertModal from './SkuBulkInsertModal'
+import { mapSkusToSelect } from '../../../../../utils/mappers'
+import getSkus from '../../../../../graphql/getSkus.graphql'
+import { SelectObject, SelectObjectProps } from '../index'
+import { SelectValue } from '../SelectObject'
+import { parseSkusFromInput, messages } from './utils'
+import IconFile from '../../../../Icon/File'
+
+export interface Props extends SelectObjectProps, WithSkusProps {
+  skus: Sku[]
+}
+
+const LabelWithTooltip: React.FC<OptionProps<any>> = props => (
+  <div {...props.innerProps}>
+    <Tooltip label={props.data.label}>
+      <span>{props.data.value}</span>
+    </Tooltip>
+  </div>
+)
+
+const SkuMultiSelectControl: React.FC<MultiValueProps<any>> = props => (
+  <components.MultiValue
+    {...props}
+    components={{
+      ...props.components,
+      Label: LabelWithTooltip,
+    }}
+  />
+)
+
+const handleInputPaste = (
+  event: React.ClipboardEvent,
+  setParsedSkus: (input: number[]) => void
+) => {
+  const inputText = event.clipboardData?.getData('text')
+  if (!inputText) return
+  const parsedSkus = parseSkusFromInput(inputText)
+  setParsedSkus(parsedSkus)
+}
+
+const SkusSelectObject: React.FC<Props> = ({
+  skus,
+  searchForSkus,
+  onChange,
+  loading = false,
+  ...props
+}) => {
+  const intl = useIntl()
+
+  const [isModalOpen, changeModalOpenState] = useState(false)
+  const [parsedSkus, setParsedSkus] = useState<number[]>([])
+
+  const { loading: skuBulkInsertLoading } = useQuery(getSkus, {
+    variables: { skus: parsedSkus },
+    skip: parsedSkus.length === 0,
+    onCompleted: data => {
+      const { skusFound, notFound } = data.getSkus
+      const values = skusFound.map(
+        (product: { id: string; fullname: string }) => ({
+          value: product.id,
+          label: `${product.id} - ${product.fullname}`,
+        })
+      )
+      const error = notFound.length
+        ? intl.formatMessage(messages.skusNotFound(notFound.join(', ')))
+        : undefined
+
+      const currentValues = props.value
+      let newValues: SelectValue[]
+      if (!currentValues) newValues = values
+      else if (Array.isArray(currentValues))
+        newValues = currentValues.concat(values)
+      else newValues = [currentValues].concat(values)
```

Comment:
```suggestion
      const newValues = currentValues ? currentValues.concat(values) : values
```

I think that the SKU Select is always multi, so you don't need to check is it is an Array. Maybe the typing is wrong, tho.

Reply:
Yep, current typing does not guarantee that 🤷‍♂

Reply:
:developers: